test/tstLinearBVH.cpp

```diff
@@ -36,6 +36,14 @@ struct TreeTypeTraits<WORKAROUND_SEQUENCE_OF_TYPES<DeviceTypes...>>
   using type = WORKAROUND_SEQUENCE_OF_TYPES<ArborX::BVH<DeviceTypes>...>;
 };
 
+template <typename... DeviceTypes>
+struct TreeTypeTraits<
+    WORKAROUND_SEQUENCE_OF_TYPES<DeviceTypes..., boost::mpl::na>>
```

Comment:
Although it looks like a really good idea, this will not work because the template parameter pack has to be the final argument